### PR TITLE
Prefs [TASK][26321] API Token settings

### PR DESCRIPTION
**Pull request by toandv98**
## Related Tickets

- [#Task 26321](https://edu-redmine.sun-asterisk.vn/issues/26321)

## WHAT

- ### API Token settings

## HOW

- Input API Token
- Save encrypted API token
- Required PIN

## Review Checklist

| Category      | View Point                                                   | Description                                                  | Expected Reviewer Answer | Reviewer1 (ToanDV) | Reviewer2 (name)   |
| ------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------ | ------------------ | ------------------ |
| General       | Code should be self-explanatory.                             | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Is this function or class too big? If yes, is the function or class having too many responsibilities? |                                                              | NO                       | <li>- [x] no</li>  | <li>- [ ] no</li>  |
| General       | Does the code follow DRY?: Do not Repeat Yourself            | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Exception handling and cleanup (dispose) resources           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Does the code build and execute correctly?                   |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Conventions   | Does the source code have end-line errors?                   |                                                              | NO                       | <li>- [x] no</li>  | <li>- [ ] no</li>  |
| Conventions   | No hard coding, use constants/configuration values           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Conventions   | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Redmine       | Does the ticket follow Sun* Redmine working process?         | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |

## Notes (Optional)

*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/edu/sun/oereminder/ui/settings/ChatWorkDialog.kt [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+    override fun getTheme() = R.style.MessageDialogStyle
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.dialog_chatwork, container, false)
+    }
```
Comment (aug16vcc) [inline app/src/main/java/com/edu/sun/oereminder/ui/settings/ChatWorkDialog.kt]:
```suggestion
    ) = inflater.inflate(R.layout.dialog_chatwork, container, false)
```

File: app/src/main/java/com/edu/sun/oereminder/ui/settings/ChatWorkDialog.kt [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                        bundleOf(RESULT_CODE to CODE_OK, RESULT_TOKEN to token)
+                    )
+                    dismiss()
+                } else {
+                    inputLayout.error = getString(R.string.msg_required_field)
+                }
+                true
+            } else {
+                false
+            }
```
Comment (aug16vcc) [inline app/src/main/java/com/edu/sun/oereminder/ui/settings/ChatWorkDialog.kt]:
```suggestion
            if (it.itemId != R.id.item_save) false
            val token = edtToken.text.toString()
            if (token.isNotBlank()) {
                setFragmentResult(
                    REQUEST_TOKEN_RESULT,
                    bundleOf(RESULT_CODE to CODE_OK, RESULT_TOKEN to token)
                )
                dismiss()
            } else {
                inputLayout.error = getString(R.string.msg_required_field)
            }
            true
```

**Review by aug16vcc — APPROVED** [review]

File: app/src/main/res/layout/dialog_chatwork.xml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <WebView
+        android:id="@+id/webViewChatWork"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        android:layout_marginTop="@dimen/dp_8"
+        android:visibility="visible"
```
Comment (huytq-2422) [inline app/src/main/res/layout/dialog_chatwork.xml]: xóa bỏ thuộc tính thừa

File: app/src/main/java/com/edu/sun/oereminder/ui/settings/ChatWorkDialog.kt [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+                        if (loadToken) {
+                            evaluateJavascript(SCRIPT_TOKEN) {
+                                AppExecutors.getInstance().mainThread.execute {
+                                    edtToken.setText(it.trimQuotationMark())
+                                }
+                            }
+                        } else {
+                            loadToken = true
+                        }
+                    }
```
Comment (huytq-2422) [inline app/src/main/java/com/edu/sun/oereminder/ui/settings/ChatWorkDialog.kt]: Dùng when(url) {...}

**Pull request merged**